hugolib/site_test.go

```diff
@@ -963,3 +963,114 @@ func TestRefLinking(t *testing.T) {
 
 	// TODO: and then the failure cases.
 }
+
+var simpleTitledPage = []byte(`+++
+title = "SimplePage"
++++`)
+
+var complexTitledPage = []byte(`+++
+title = "Complex Page: More Complicated Title / With Special Stuff"
++++`)
+
+func setupPageTitlesMockSite(t *testing.T) *Site {
+	sources := []source.ByteSource{
+		{Name: filepath.FromSlash("simple.md"), Content: simpleTitledPage},
+		{Name: filepath.FromSlash("complex.md"), Content: complexTitledPage},
+	}
+
+	cfg, fs := newTestCfg()
+
+	cfg.Set("baseURL", "http://auth")
+
+	writeSourcesToSource(t, "content", fs, sources...)
+	return buildSingleSite(t, deps.DepsCfg{Fs: fs, Cfg: cfg}, BuildCfg{})
+}
+
+func TestGetPageByTitle(t *testing.T) {
+	t.Parallel()
+	site := setupPageTitlesMockSite(t)
+
+	for i, test := range []struct {
+		pageKind          string
+		pageTitle         string
+		expectedPermalink string
+	}{
+		{KindPage, "SimplePage", "http://auth/simple/"},
+		{KindPage, "Complex Page: More Complicated Title / With Special Stuff", "http://auth/complex/"},
+	} {
+		// This call to GetPage simulates a template invocation such as:
+		// $.Site.GetPage "pageKind" "pageTitle"
+		p, err := site.Info.GetPage(test.pageKind, test.pageTitle)
```

Comment:
Where does it say that we support page title in `GetPage`? Didn't we agree previously that this was ambigous and not possible?

Reply:
Hi @bep ,

In #4092 , when discussing how to formulate a link to a taxonomy term (given the un-sanitized form of the term), you suggested using GetPage like this:

> > Is there a better way to generate a permalink to a specific term's page?
>
> ```
> {{ .Site.GetPage "taxonomy" "game" "myterm" }}
> ```

What these tests demonstrate is that this will only work if the term does not have an associated content file.  If there is an associated content file, then the term's Page is indexed by the sanitized path to that file, instead.

If it isn't the intent of GetPage to retrieve pages by "title" - and the fact that this work for terms without a content file, is just a coincidence - then my request for "a better way to generate a permalink to a specific term's page" is still outstanding.

Reply:
Sure, but then you should create a test with a term with and without a content file and do 

 .Site.GetPage "taxonomy" "game" "termWithContentFile"
 .Site.GetPage "taxonomy" "game" "termWithOutContentFile"

Not some totally unrelated test.

Reply:
If you look a few lines down, that's what `TestGetPageByTaxonomyTerm` tests.

The `TestGetPageByTitle` test is included because, based on my code investigation in #4118 , these two behaviors are very closely related.  They both present the same "problem" of retrieving a Page by its Title; the latter case just happens to be Titled with a taxonomy term.

As a user, I would find it odd if GetPage was fixed to allow retrieving a taxonomy term's Page by Title, but not other types of Page.

That said, since you're right that it isn't the specific focus of the bugs I've been discussing, I'll update my test PR to remove this distraction.